### DISCUSSION TITLE
Share interface : Connection pool sharing between threads

### DISCUSSION THREAD

**Discussion by MatouCode** [curl/curl#12895] [Q&A] [answered]
Hello, 

I saw the video "[mastering libcurl](https://youtu.be/9KqnXsSxqGA?feature=shared&t=4514)", in which it says on the share interface that it cannot reuse the connection cache between multiple threads, and that this is a known bug. 

As I don't see it in the [libcurl's "Known Bugs" page](https://curl.se/docs/knownbugs.html), and I don't see anything about it in the [changelog ](https://curl.se/changes.html) since 8.4.0, I was wondering if the bug had been fixed or if the feature had been removed. This is not clear to me because the [share interface overview ](https://curl.se/libcurl/c/libcurl-share.html)still mentions the connection cache reuse. 

Thanks,

**Comment by bagder** [comment, marked as answer]
## 1

In the manpage for [CURLSHOPT_SHARE](https://curl.se/libcurl/c/CURLSHOPT_SHARE.html), where  `CURL_LOCK_DATA_CONNECT` is documented, it says:

> It is not supported to share connections between multiple concurrent threads. 

## 2

In the TODO file, item "1.4" under [Better and more sharing](https://curl.se/docs/todo.html#Better_and_more_sharing) it says:

>  The share interface offers CURL_LOCK_DATA_CONNECT to have multiple easy
 handle share a connection cache, but due to how connections are used they are
 still not thread-safe when used shared.

> See https://github.com/curl/curl/issues/4915 and lib1541.c
